gensim/test/test_word2vec.py

```diff
@@ -44,19 +44,33 @@ def __iter__(self):
     ['graph', 'minors', 'survey']
 ]
 
+new_sentences = [
+    ['computer', 'artificial', 'intelligence'],
+    ['artificial', 'trees']
+]
 
 def testfile():
     # temporary data will be stored to this file
     return os.path.join(tempfile.gettempdir(), 'gensim_word2vec.tst')
 
-
 class TestWord2VecModel(unittest.TestCase):
+
     def testPersistence(self):
         """Test storing/loading the entire model."""
         model = word2vec.Word2Vec(sentences, min_count=1)
         model.save(testfile())
         self.models_equal(model, word2vec.Word2Vec.load(testfile()))
 
+    def testOnlineLearning(self):
+        """Test that the algorithm is able to add new words to the
+        vocabulary and to a trained model"""
+        model = word2vec.Word2Vec(sentences, min_count=1)
+        model.update_vocab(new_sentences)
+        model.train(new_sentences)
+        self.assertEqual(len(model.vocab), 14)
+        self.models_equal(model, word2vec.Word2Vec.load(datapath(
+            "gensim_word2vec_update.tst")))
```

Comment:
Well, you can get inspiration from the other surrounding tests :)

But here I meant: where does the file `gensim_word2vec_update.tst` come from? I don't see it in the repository.

And this test should probably be finer anyway -- relying on bit-for-bit equality against a pregenerated model seems to brittle. Any future change to the RNG, or any other parameter, will result in a fail. My guess is this won't work even now, across different Python version -- did you run the tests pass on Python 2.6 & 2.7 & 3.4? Once the conflicts are resolved, Github will launch Travis tests and we will be able to see the test results.